tools/deps_info.py

```diff
@@ -57,14 +57,12 @@
   'SDL_OpenAudio': ['malloc', 'free'],
   'SDL_PushEvent': ['malloc', 'free'],
   'SDL_free': ['free'],
-  'SDL_getenv': ['malloc', 'free'],
   'SDL_malloc': ['malloc', 'free'],
   '__ctype_b_loc': ['malloc', 'free'],
   '__ctype_tolower_loc': ['malloc', 'free'],
   '__ctype_toupper_loc': ['malloc', 'free'],
-  '__cxa_allocate_exception': ['malloc', 'free', 'setThrew'],
```

Comment:
I think the original thinking here was that if you don't allocate an exception then you can't do almost anything with exceptions. So one entry for the allocate function could add in deps for all the others. That is, we may be missing entries for more libcxxabi things, and the change here (that removes from allocate) may force us to need them now.

Similiar thinking was applied on `SDL_Init` iirc, and others - add in a bunch of deps for a key function that gates all the others.

But maybe I'm missing something and this is not an issue?

Reply:
I'd rather be precise here if we can, especially since it allows us to verify that a given dependency really is valid.

If we care about code size we should care about being precise here.  

Perhaps that other example are more compelling but this one seems very clear cut.
`__cxa_end_catch` is the function that depends on free and `__cxa_allocate_exception` is the one that depends on malloc.   Having this additional dependency of `__cxa_allocate_exception` on `free` doesn't make any sense does it?   Because its a duplicate .. there is no benefit at all.    Its redundant, inaccurate and confusing.

`setThrew` only has a single caller which is  __cxa_end_catch so I think this part of the change is clearly correct by inspection.

   `

Reply:
I can't find any active examples of the kind of thing you are describing.   I.e. a gatekeeper function having a fake dependency (one that it doesn't actually depend on) in order to avoid added that dependency to N other functions that the gatekeeper would normally be used with but doesn't actually depend on.    

Even if we did that in the past (or perhaps there is an example here of that that I'm missing) I would still rather fix it by being precise in the way that this test enforces.

And even if this test does result in some undefined symbol errors (I don't think it will but its possible) it should be easy to diagnose them and to add the precise dependencies.

Reply:
I agree being precise is better. And hopefully there is no issue here, but it does worry me that imprecision in the past could make it harder to notice breaking anything now. I'm probably overly worried though.